.sphinx/lxddoc/README.md

```diff
@@ -0,0 +1,129 @@
+		}
+
+		err := validator(v)
+		if err != nil {
+			return fmt.Errorf("Invalid cluster configuration key %q value", k)
+		}
+	}
+
+	return nil
+```
+
+The go-swagger spec from source generator can only handles `swagger:meta` (global file/package level documentation), `swagger:route` (API endpoints), `swagger:params` (function parameters), `swagger:operation` (method documentation), `swagger:response` (API response content documentation), `swagger:model` (struct documentation) generation. In our use case, we would want a config variable spec generator that can bundle any key-value data pairs alongside metadata to build a sense of hierarchy and identity (we want to associate a unique key to each lxddoc comment group that will also be displayed in the generated documentation)
+
+In a swagger fashion, `lxddoc` can associate metadata key-value pairs (here for example, `group` and `key`) to data key-value pairs. As a result, it can generate a YAML tree out of the code documentation and also a markdown document.
+
+## Output
+
+Here is the YAML output of the example shown above:
+
+```yaml
+configs:
+    cluster:
+        - scheduler.instance:
+            condition: container
+            default: '`all`'
+            liveupdate: '`yes`'
+            longdesc: |4-
+
+                 The cluster config keys
+            shortdesc: Possible values are all, manual and group. See Automatic placement of instances for more information.
+            type: integer
+        - user.*:
+            condition: container
+            default: '-'
+            liveupdate: '`yes`'
+            longdesc: |4+
+
+                This is the real long desc.
+
+                With two paragraphs.
+
+                And a list:
+
+                - Item
+                - Item
+                - Item
+
+                And a table:
+
+                Key                                 | Type      | Scope     | Default                                          | Description
+                :--                                 | :---      | :----     | :------                                          | :----------
+                `acme.agree_tos`                    | bool      | global    | `false`                                          | Agree to ACME terms of service
+                `acme.ca_url`                       | string    | global    | `https://acme-v02.api.letsencrypt.org/directory` | URL to the directory resource of the ACME service
+                `acme.domain`                       | string    | global    | -                                                | Domain for which the certificate is issued
+                `acme.email`                        | string    | global    | -                                                | Email address used for the account registration
+
+            shortdesc: Free form user key/value storage (can be used in search).
+            type: string
+
+```
+
+To see the mardown output, see `.sphinx/lxddoc/lxddoc.txt`
```

Comment:
```suggestion
To see the Markdown output, see `.sphinx/lxddoc/lxddoc.txt`
```
(There's a few more "Markdown"s further up, but the capitalisation isn't really that important :wink: )

Might be better to include the output here?
I don't think we need an extra file for that, we already have `config_options_cheat_sheet.md` with the same syntax.